lib/rules/selector-anb-no-unmatchable/__tests__/index.js

```diff
@@ -0,0 +1,145 @@
+'use strict';
+
+const { messages, ruleName } = require('..');
+
+testRule({
+	ruleName,
+	config: true,
+
+	accept: [
+		{
+			code: 'a:nth-child(1)',
+		},
+		{
+			code: 'a:nth-child(0n+1)',
+		},
+		{
+			code: 'a:nth-child(0n-1)',
+		},
+		{
+			code: 'a:nth-child(2n+0)',
+		},
+		{
+			code: 'a:nth-child(2n-0)',
+		},
+		{
+			code: 'a:nth-child(1 of a) {}',
+		},
+		{
+			code: 'a:nth-last-child(1)',
+		},
+		{
+			code: 'a:nth-of-type(1)',
+		},
+		{
+			code: 'a:nth-last-of-type(1)',
+		},
+	],
+
+	reject: [
```

Comment:
question

What happens with https://www.w3.org/TR/css-syntax-3/#example-3d25855e?

Reply:
it errors; I'll add some test cases for this!